plugins/aws-catalog-backend/src/RDSEntityProvider.ts

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright SeatGeek
+ * Licensed under the terms of the Apache-2.0 license. See LICENSE file in project root for terms.
+ */
+import {
+  DBInstance,
+  DescribeDBInstancesCommandInput,
+  RDSClient,
+  paginateDescribeDBInstances,
+} from '@aws-sdk/client-rds';
+import { AwsCredentialIdentity } from '@aws-sdk/types';
+import { TaskRunner } from '@backstage/backend-tasks';
+import { Config } from '@backstage/config';
+import { Logger } from 'winston';
+import {
+  BaseEntityProvider,
+  InstanceFilter,
+  InstanceTransformer,
+} from './BaseEntityProvider';
+
+/**
+ * A db instance from the AWS SDK.
+ * @public
+ */
+export type RDSDBInstance = DBInstance;
+/**
+ * Input for the RDSDescribeDBInstancesCommand.
+ * @public
+ */
+export type RDSDescribeDBInstancesCommandInput =
+  DescribeDBInstancesCommandInput;
+
+/**
+ * Discovers RDS instances from AWS and adds them to the catalog.
+ * @public
+ */
+export class RDSEntityProvider extends BaseEntityProvider<RDSDBInstance> {
+  private readonly client: RDSClient;
+  private readonly commandInput: RDSDescribeDBInstancesCommandInput;
+
+  static fromConfig(
+    config: Config,
+    options: {
+      logger: Logger;
+      schedule: TaskRunner;
+      transformer: InstanceTransformer<RDSDBInstance>;
+      filter?: InstanceFilter<RDSDBInstance>;
+      commandInput?: RDSDescribeDBInstancesCommandInput;
+    },
+  ): RDSEntityProvider[] {
+    const awsConfig = config.getOptionalConfig('catalog.providers.aws');
+    if (!awsConfig) {
+      options.logger.warn(
+        "RDSEntityProvider will not be created as 'catalog.providers.aws' key is missing in configuration.",
+      );
+      return [];
+    }
+    return awsConfig.keys().map(key => {
+      const individalConfig = awsConfig.getConfig(key);
+      const region = individalConfig.getString('region');
+      const accessKeyId = individalConfig.getOptionalString('accessKeyId');
+      const sessionToken = individalConfig.getOptionalString('sessionToken');
+      const secretAccessKey =
+        individalConfig.getOptionalString('secretAccessKey');
+      let credentials: AwsCredentialIdentity | undefined = undefined;
+      if (accessKeyId && secretAccessKey) {
+        credentials = {
+          accessKeyId,
+          secretAccessKey,
+          sessionToken,
+        };
+      }
```

Comment:
once we add more integratons we should look into how we can genericize this as well

Reply:
I've taken a stab at this via 4d95ac2.

Reply:
would be interesting if at some point we just create one AWS entity provider in code and then decide which integrations get enabled via config, but let's save that til later